rdata/conversion/_conversion.py

```diff
@@ -820,6 +872,9 @@ def _convert_next(  # noqa: C901, PLR0912, PLR0915
 
             value = None
 
+        elif obj.info.type == parser.RObjectType.ALTREP:
+            value = convert_altrep_to_range(obj)
```

Comment:
I think altreps are intended to be an internal representation detail. That is why I made the default to expand them in the parser: I assumed that, except for very specific situations, users would prefer to ignore their existence.

That said, IMHO the default conversion routines should be able to deal with altreps (in case that `expand_altreps`  is `False`). I still think they should convert it to the same object as if they weren't altreps, for consistency.

For me ranges are a very different beast than NumPy arrays, and returning such different kind of objects for the "same" underlying R object would be confusing. Moreover, there is no equivalent to `range` for compact sequences of float.

If we want to convert an array back to an altrep for space savings we could probably try to detect if the array is a sequence (with `np.diff`, a subtraction, and `np.all` or something like that it would be easy, at least for integer dtype). However that would convert to altrep even arrays that were **not** stored as altreps in the original file, for whatever reason. So, the choice would be between no altreps at all when writing, or all possible ones.

For the above reasons I think it is preferrable **not** to consider altreps in the roundtrip comparison (that is, comparing both files with `expand_altreps=True`. Any thoughts on this?